### PR TITLE
(fix) vnode._children should keep their type as array when diffing

### PR DESCRIPTION
**Pull request by cristianbote**
When trying to pave the way for `preact-redux` with the latest rc, seems that the `_children` is loosing it's array type whenever `react-redux` is trying to `.forceUpdate()`.
This is the relevant error and log https://travis-ci.org/developit/preact-redux/builds/583356189#L390

Adds `+2B`

### CODE REVIEW

**Comment by cristianbote** [comment]
I still need to add a test for this.

File: src/diff/index.js [hunk 1]
```diff
@@ -118,7 +118,7 @@ export function diff(parentDom, newVNode, oldVNode, context, isSvg, excessDomChi
 
 			tmp = c.render(c.props, c.state, c.context);
 			let isTopLevelFragment = tmp != null && tmp.type == Fragment && tmp.key == null;
-			newVNode._children = isTopLevelFragment ? tmp.props.children : tmp;
+			newVNode._children = toChildArray(isTopLevelFragment ? tmp.props.children : tmp);
```
Comment (developit) [inline src/diff/index.js]: nit: should we pass the Array directly? (`toChildArray(.., [])`)

Comment (cristianbote) [reply src/diff/index.js]: Well the third argument of `toChildArray` is actually the returned flattened array, and if that's missing is defined inside the function and returned. Was thinking that without passing it we could save some bytes.

**Comment by marvinhagemeister** [comment]
@cristianbote did you manage to find a test case for this?👀 I had in mind that it would be added before merging, but my comment wasn't clear at all

**Comment by cristianbote** [comment]
Oh 💩 totally forgot about it! Will add it asap. Thanks for reminding me! 😟

**Pull request merged**